### PR TITLE
Add reference repo updater pipeline

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Updates all reference repos on all slaves
- Excludes master, proxy and zOS since they don't work

Related #5195
[skip ci]

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
@janvrany would you mind reviewing this as well since you have pipeline experience.
Here is my test job
https://ci.eclipse.org/omr/job/Ref_Repo_Updater/

**Comment by 0xdaryl** [comment]
Apologies if I'm being obtuse, but can you provide an explanation for what this is for and the effect on our CI testing if it is committed?

**Comment by janvrany** [comment]
@AdamBrousseau : oops, I'm very sorry - I believed I did give my LGTM to this PR weeks ago but apparently I did not. 
So: looks good. 

@0xdaryl : in PR #5195, Adam changed the code to use "reference repository" when checking out the code. This repository is stored at build machines - the idea is - roughly - that you first pull commits from that reference repo and then you pull what's missing from github. This way, you save bandwidth as - ideally - you pull only "few" commits over network. In other words, this reference repo is just a locally stored cache. 

This reference repo should be updated "time to time" to contain to keep it reasonably up-to-date with github (so when building, one only pull day (week, month, ...) worth of commits). 

It should be mentioned that "build" pipelines work regardless this "reference repo" is present or not - if missing, everything is pulled from github. 

This "reference repo updater" is just a pipeline that creates and/or updates that reference repo on all build machines. The idea is to run this pipeline (automatically) every day (week, month, ...) to keep this reference repo reasonably up-to-date. 

@AdamBrousseau, please correct me if I'm wrong.

**Comment by AdamBrousseau** [comment]
@janvrany Very good explanation. Thanks for the review. Here is a link to the job I had setup which is currently scheduled `@monthly`.
https://ci.eclipse.org/omr/job/Ref_Repo_Updater/

File: buildenv/jenkins/refRepoUpdater.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2209, 2020 IBM Corp. and others
```
Comment (youngar) [inline buildenv/jenkins/refRepoUpdater.groovy:2]:
```suggestion
 * Copyright (c) 2020, 2020 IBM Corp. and others
```

Comment (AdamBrousseau) [reply buildenv/jenkins/refRepoUpdater.groovy:2]: It's ok, I am from the future.

**Comment by youngar** [comment]
@AdamBrousseau Is this good to go?

**Comment by AdamBrousseau** [comment]
Yes I believe so. It has been running in production out of my fork with no issues. Once merged, I will update the job to point at eclipse/omr.

**Comment by youngar** [comment]
Thanks @AdamBrousseau. Sorry for the long delay!

**Pull request merged**